test/stubs/dummy_project_1/dummy_project_1/expected.json

```diff
@@ -78,74 +97,170 @@
             "name": "Microsoft.Diagnostics.Tracing.EventSource.Redist",
             "version": "1.1.28",
             "targetFramework": ".NETFramework4.0",
+            "dependencies": {},
+            "versionSpec": "1.1.28",
+            "from": [
+              "Microsoft.Diagnostics.Tracing.EventSource.Redist@1.1.28"
+            ]
+          },
+          "Microsoft.Bcl.Async": {
+            "name": "Microsoft.Bcl.Async",
+            "version": "1.0.168",
+            "targetFramework": "WindowsPhone8.0",
+            "dependencies": {},
+            "versionSpec": "1.0.168",
+            "from": [
+              "Microsoft.Bcl.Async@1.0.168"
+            ]
+          }
+        },
+        "versionSpec": "2.2.0",
+        "from": [
+          "dummy_project_1@0.0.0",
+          "Microsoft.ApplicationInsights@2.2.0"
+        ]
+      },
+      "Microsoft.ApplicationInsights.Agent.Intercept": {
+        "name": "Microsoft.ApplicationInsights.Agent.Intercept",
+        "version": "2.0.6",
+        "targetFramework": "net461",
+        "dependencies": {
+          "Microsoft.Diagnostics.Tracing.EventSource.Redist": {
+            "name": "Microsoft.Diagnostics.Tracing.EventSource.Redist",
+            "version": "1.1.28",
+            "targetFramework": ".NETFramework4.0",
+            "dependencies": {},
+            "versionSpec": "1.1.28",
+            "from": [
+              "Microsoft.Diagnostics.Tracing.EventSource.Redist@1.1.28"
```

Comment:
this seems wrong ... should contain the whole pass to the root. Same happens with more trees bellow